app/models/facet_collection.rb

```diff
@@ -20,6 +20,16 @@ def filters
     facets.select(&:filterable?)
   end
 
+  def ensure_minimum_one_open_facet(facets_to_ensure)
+    at_least_one_facet_selected = (facets_to_ensure.map { |f| f.open_on_load? }).any?
```

Comment:
Pedantic, but you can pass `any?` a block and thus refactor this to be `facets_to_ensure.any? { |f| f.open_on_load? }` which is slightly shorter (probably also a tiny, tiny bit more efficient, although I've probably wasted more CPU cycles and time then this will ever save by typing this comment.
https://apidock.com/ruby/Enumerable/any%3F